src/services/link/index.js

```diff
@@ -16,25 +16,49 @@ module.exports = function setup() {
 		lean: true,
 	};
 
+	const registrationLinkExpirationDays = app.get('registrationLinkExpirationDays') || 30;
+
 	let linkService = service(options);
 
-	function redirectToTarget(req, res, next) {
-		if (req.method === 'GET' && !req.query.target) { // capture these requests and issue a redirect
+	const hasExpired = (date) => {
+		// calculate the date the link will expire on
+		const expirationDate = new Date(date);
+		expirationDate.setDate(expirationDate.getDate() + registrationLinkExpirationDays);
+		// check if that expiration date has already passed
+		return expirationDate <= Date.now();
+	};
+
+	const isLocalRegistrationLink = (url) => {
+		// calculate the beginning of a registrationLink e.g. https://schul-cloud.org/registration/
+		const linkPrefix = `${app.settings.services.web}/registration/`;
+		// check if url starts with that prefix
+		return (url.startsWith(linkPrefix));
+	};
+
+	function checkLink(req, res, next) {
+		if (req.method === 'GET' && !req.query.target) {
+			// capture queries that don't look for the target and check them
+			/* eslint-disable-next-line no-underscore-dangle */
 			const linkId = req.params.__feathersId;
 			linkService.get(linkId)
 				.then((data) => {
-					if (data.data || req.query.includeShortId) {
+					if (isLocalRegistrationLink(data.target) && hasExpired(data.createdAt)) {
+						// link is both local and has expired
+						res.status(403).json({ error: 'Expired' });
+					} else {
+						// link is valid
 						const [url, query] = data.target.split('?');
 						const queryObject = queryString.parse(query || '');
 						queryObject.link = data._id;
-						res.redirect(`${url}?${queryString.stringify(queryObject)}`);
-					} else {
-						res.redirect(data.target);
+						res.json({
+							target: `${url}?${queryString.stringify(queryObject)}`,
+						});
 					}
 				})
 				.catch((err) => {
-					logger.warning(err);
-					res.status(500).send(err);
+					logger.error(err);
+					// send not found to client
+					res.status(404).json({ error: 'Not found' });
```

Comment:
you may return res.foo() and next() statements to make it clear that nothing happens afterwards